packages/eslint-plugin-pf-codemods/src/rules/helpers/renameComponent.ts

```diff
@@ -0,0 +1,117 @@
+import { Rule } from "eslint";
+import { getAllImportsFromPackage } from "./getFromPackage";
+import {
+  ImportSpecifierWithParent,
+  ImportDefaultSpecifierWithParent,
+  JSXOpeningElementWithParent,
+} from "./interfaces";
+import {
+  getDefaultDeclarationString,
+  getComponentImportName,
+  getNodeName,
+  hasCodeModDataTag,
+} from "./index";
+
+interface ComponentRenames {
+  [currentName: string]: string;
+}
+
+function formatDefaultMessage(oldName: string, newName: string) {
+  return `${oldName} has been renamed to ${newName}.`;
+}
+
+function getFixes(
+  fixer: Rule.RuleFixer,
+  nodeImport: ImportSpecifierWithParent | ImportDefaultSpecifierWithParent,
+  node: JSXOpeningElementWithParent,
+  oldName: string,
+  newName: string
+) {
+  const fixes = [];
+
+  const isNamedImport = nodeImport.type === "ImportSpecifier";
+  const importDeclaration = nodeImport.parent;
+  const importSource = importDeclaration?.source.raw;
+  const importSourceHasComponentName = importSource?.includes(oldName);
+  const newImportDeclaration = importSource?.replace(oldName, newName);
+
+  if (isNamedImport) {
+    fixes.push(fixer.replaceText(nodeImport.imported, newName));
+  }
+
+  if (
+    importDeclaration &&
+    newImportDeclaration &&
+    importSourceHasComponentName
+  ) {
+    fixes.push(
+      fixer.replaceText(importDeclaration.source, newImportDeclaration)
+    );
+  }
+
+  const shouldRenameNode =
+    isNamedImport && nodeImport.imported.name === nodeImport.local.name;
+
+  if (shouldRenameNode) {
+    fixes.push(fixer.replaceText(node.name, newName));
+    fixes.push(fixer.insertTextAfter(node.name, " data-codemods"));
```

Comment:
I wonder if this should be optional, I know we need the `data-codemods` for the followup Masthead rule, but in other cases it may be an unnecessary addition.

But if anything, it can be done as a followup.

Reply:
Yeah I went back and forth on that and IIRC talked it over with @thatblindgeye, I think we landed on always adding it to prevent issues and adding https://github.com/patternfly/pf-codemods/issues/723 so that we aren't leaving consumer codebases littered with our data tags.

I'm not super opposed to making it optional though, but yeah would want to do it as a followup.